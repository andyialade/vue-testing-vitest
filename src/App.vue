<script setup>
import { ref } from 'vue';
import { getConcerts } from './services/concert-service';
import Concerts from './components/Concerts.vue'
const items = ref([]);
(async function(){
  items.value = await getConcerts()
})()

</script>



<template>
  <!-- <h1>Portal</h1> -->
  <h1>Musician's World</h1>
  <Concerts 
    :items="items" 
  />
</template>