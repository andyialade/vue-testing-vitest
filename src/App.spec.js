import { describe, it, expect } from "vitest";
import App from "./App.vue"
import { createApp } from "vue";
import { renderToString } from "@vue/server-renderer";


describe("the application", () => {
    it("should not regress (snapshot test)", async() => {
        const app = createApp(App)
        const string = await renderToString(app)
        expect(string).toMatchSnapshot()
    })
})