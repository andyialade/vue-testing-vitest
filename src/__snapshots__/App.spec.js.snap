// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`the application > should not regress (snapshot test) 1`] = `"<!--[--><!-- <h1>Portal</h1> --><h1>Musicians World</h1><!--[--><h1> Musician Zone </h1><button> Concerts </button><h2> Your Concerts </h2><div><div><!--v-if--><!--[--><!--]--></div></div><h2> Wishlist </h2><div><div><!--v-if--><!--[--><!--]--></div></div><!--]--><!--]-->"`;
